internal/firmware/info.go

```diff
@@ -0,0 +1,63 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+/*
+ * Package firmware extracts information from the running system by
+ * inspecting metadata files embedded in the root filesystem.
+ */
+package firmware
+
+// Info provides information about firmware.
+type Info struct {
+	Version         string
+	OriginalName    string
+	OriginalSummary string
+
+	StoreInfo
```

Comment:
I don't really understand why `StoreInfo` is its own type, rather than the fields just listed here. It seems it's not used elsewhere (except in one test).

Reply:
I will update the comments to explain this, but the historical reason is that the store attributes `StoreInfo` are used as a transactional handle to locate a specific firmware, and is used everywhere in the code in different packages. The wider `Info` structure includes lots of additional information which are not replicated in the state, but rather looked up if needed (some of the `Info` attributes are overridden by `StoreInfo` store approved versions).

Reply:
Had a discussion. Fixed.